driver-core/src/main/java/com/datastax/driver/core/Responses.java

```diff
@@ -421,8 +421,10 @@ public String toString() {
                             sb.append(" | ");
                             if (metadata.columns == null)
                                 sb.append(Bytes.toHexString(v));
-                            else
-                                sb.append(metadata.columns.getType(i).deserialize(v));
+                            else {
+                                // using default codec registry as this is only intended for debugging purposes
+                                sb.append(CodecRegistry.DEFAULT_INSTANCE.codecFor(metadata.columns.getType(i)).deserialize(v));
```

Comment:
It's just for debugging purposes so I think it's fine to use the default instance here.